File: fhg-api/src/main/java/de/jbamberger/api/FhgApiComponent.java
package de.jbamberger.api;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */
@Singleton
@Component(modules = {
        FhgApiModule.class,
        NetModule.class
})
public interface FhgApiComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        FhgApiComponent build();
    }

    void inject(FhgApi.Provider builder);
}

File: fhg-api/src/main/java/de/jbamberger/api/FhgApiComponent.kt
package de.jbamberger.api

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */
@Singleton
@Component(modules = [(FhgApiModule::class), (NetModule::class)])
interface FhgApiComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): FhgApiComponent
    }

    fun inject(builder: FhgApi.Provider)
}
